File: app/src/main/java/com/vladislav/workoutassistant/ui/exercises/viewmodels/ExerciseViewModel.java
package com.vladislav.workoutassistant.ui.exercises.viewmodels;

import android.app.Application;

import com.vladislav.workoutassistant.data.DataRepository;
import com.vladislav.workoutassistant.data.db.entity.Exercise;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MediatorLiveData;
import androidx.lifecycle.Observer;

public class ExerciseViewModel extends AndroidViewModel {

    private DataRepository mDataRepository;
    private MediatorLiveData<Exercise> mExercise = new MediatorLiveData<>();

    public ExerciseViewModel(Application application) {
        super(application);
        mDataRepository = DataRepository.Companion.getInstance(application);
    }

    public void init(final int exerciseId) {
        mExercise.addSource(mDataRepository.loadExercise(exerciseId), new Observer<Exercise>() {
            @Override
            public void onChanged(Exercise exercise) {
                mExercise.setValue(exercise);
            }
        });
    }

    public LiveData<Exercise> getExercise() {
        return mExercise;
    }
}

File: app/src/main/java/com/vladislav/workoutassistant/ui/exercises/viewmodels/ExerciseViewModel.kt
package com.vladislav.workoutassistant.ui.exercises.viewmodels

import android.app.Application
import androidx.lifecycle.*
import com.vladislav.workoutassistant.data.DataRepository
import com.vladislav.workoutassistant.data.db.entity.Exercise

class ExerciseViewModel(application: Application) : AndroidViewModel(application) {

    private val mDataRepository: DataRepository = DataRepository.getInstance(application)
    private val mExerciseId = MutableLiveData<Int>()

    val exercise: LiveData<Exercise> = Transformations.switchMap(mExerciseId) {
        id -> mDataRepository.loadExercise(id)
    }

    fun loadExerciseById(id: Int) {
        mExerciseId.value = id
    }
}
